src/test/java/com/redhat/devtools/lsp4ij/LSPIJUtils_toOffsetTest.java

```diff
@@ -62,7 +63,7 @@ public void testInvalidCharacterWithOneLine() {
     }
 
     public void testInvalidCharacterWithTwoLines() {
-        assertOffset("foo\nbar", 0, 999999, 3, null);
+        assertOffset("foo\nbar", 0, 999999, 7, null);

```

Comment:
Had to change this expected value because it's no longer constrained to the line but rather to the document.

Reply:
Mmmm this test comes from vscode if I rember.

Reply:
The specs says 

`If the character value is greater than the line length it defaults back to the line length. `

https://github.com/microsoft/vscode-languageserver-node/blob/d810d51297c667bd3a3f46912eb849055beb8b6b/textDocument/src/main.ts#L40C5-L40C7

This test set line to 0 and in this case line length is equals to 3, right?

I have not found a test with this usecase. But perhaps the spec is not clear and we should  adopt your fix. The thing like I love in your PR i sthat you avoid calling getLineEndOffset.

I am not against to merge your PR,but before could you create an issue https://github.com/microsoft/vscode-languageserver-node/ with this usecase toclarify this usecase please.

Reply:
I logged this to try to track down how this is working:

https://github.com/forcedotcom/salesforcedx-vscode-slds/issues/141